roles/sap_netweaver_preconfigure/tasks/sapnote/assert-3119751.yml

```diff
@@ -5,46 +5,90 @@
     msg: "SAP note {{ (__sap_netweaver_preconfigure_sapnotes_versions | selectattr('number', 'match', '^3119751$') | first).number }}
           (version {{ (__sap_netweaver_preconfigure_sapnotes_versions | selectattr('number', 'match', '^3119751$') | first).version }}): Linux Requirements for SAP Kernel 754 and for SAP Kernel 788 and higher"
 
-- name: Get info about the compat-sap-c++-10.so file
+# Note: This file is only included for RHEL 8, so no further when condition is required here.
+
+- name: Identify all 'compat-sap-c++-NUM.so' files with NUM >= 10
+  ansible.builtin.find:
+    paths: '/opt/rh/SAP/lib64'
+    patterns: "compat-sap-c++-1*"
+  register: __sap_netweaver_preconfigure_register_find_compat_sap_cpp
+
+- name: Set fact for file names from find result
+  ansible.builtin.set_fact:
+    __sap_netweaver_preconfigure_fact_compat_sap_cpp_filenames: "{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_filenames | d([]) + [__sap_netweaver_preconfigure_item.path | basename] }}"
+  loop: "{{ __sap_netweaver_preconfigure_register_find_compat_sap_cpp.files }}"
+  loop_control:
+    loop_var: __sap_netweaver_preconfigure_item
+    label: "{{ __sap_netweaver_preconfigure_item.path }}"
+  when: __sap_netweaver_preconfigure_register_find_compat_sap_cpp.matched > 0
+
+- name: Set fact for the latest 'compat-sap-c++.NUM.so' file name
+  ansible.builtin.set_fact:
+    __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest: "{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_filenames | sort | last }}"
+  when: __sap_netweaver_preconfigure_register_find_compat_sap_cpp.matched > 0
```

Comment:
Same suggestion as above applies to these 2 tasks. They can be combined to one simpler task, since the whole purpose of the first large task is to parse the find results in a loop, which can be achieved by using the `map` filter in combination with the filters in the second task.

Reply:
again, very useful, thanks!